privacyidea/lib/token.py

```diff
@@ -1771,15 +1779,15 @@ def copy_token_user(serial_from, serial_to):
     """
     tokenobject_from = get_one_token(serial=serial_from)
     tokenobject_to = get_one_token(serial=serial_to)
-    user_id = tokenobject_from.token.user_id
-    resolver = tokenobject_from.token.resolver
-    resolver_type = tokenobject_from.token.resolver_type
-    tokenobject_to.set_user_identifiers(user_id, resolver,
-                                                resolver_type)
+    TokenOwner(token_id=tokenobject_to.token.id,
+               user_id=tokenobject_from.token.first_owner.user_id,
+               realm_id=tokenobject_from.token.first_owner.realm_id,
+               resolver=tokenobject_from.token.first_owner.resolver).save()
```

Comment:
If I understand correctly, if ``tokenobject_to`` had an owner before the copy operation, it will have two owners after the copy operation. I guess we should first unassign ``tokenobject_to`` to rule out that case?

Reply:
I think this would be half of  the intented behaviour.
I would actually expect to copy *all* users of the old token to the new token and then the new token have the old users and the new users.

Currently this function is only used in *losttoken*, where the destination token does not have any users anyways.

The destination token loosing all the existing users would not be the expected behaviour. After all it is ``copy_token_users`` and not ``replace_token_users``.

Reply:
I think the copying behavior instead of replacing makes total sense, once we allow a token to have multiple owners. :-)
Right now, however, ``copy_token_users`` it is also used in the ``/token/copyuser`` endpoint. With this PR, if I do
```
$ http POST http://localhost:5000/token/copyuser from=OATH0001E293 to=OATH0003E7E7 PI-Authorization:$TOKEN
```
the token ``OATH0003E7E7`` actually has two owners:
```python
>>> get_one_token(serial="OATH0003E7E7").token.owners.all()
[<TokenOwner 5>, <TokenOwner 6>]
```
so I think it makes more sense to have ``copy_token_users`` *replace* the owner of the destination token.

Reply:
OK, will do. (Since /token/assign) also only allows to assign a token without a user.
The old user will be removed (although usually there will never be an old user;)
(Dont forget to add a TODO/FIXME for Version 3.1)

Will do  2019-02-14

Reply:
done in 49f1ac75